app/src/main/java/org/stepik/android/presentation/lesson_demo/LessonDemoFeature.kt

```diff
@@ -4,14 +4,16 @@ import org.stepik.android.domain.course_payments.model.DeeplinkPromoCode
 import org.stepik.android.model.Course
 import org.stepik.android.presentation.course_purchase.model.CoursePurchaseData
 import org.stepik.android.presentation.course_purchase.model.CoursePurchaseDataResult
+import org.stepik.android.presentation.wishlist.WishlistOperationFeature
 
 interface LessonDemoFeature {
     sealed class State {
         object Idle : State()
-        object Loading : State()
-        object Error : State()
-        object Unavailable : State()
-        data class Content(val deeplinkPromoCode: DeeplinkPromoCode, val coursePurchaseData: CoursePurchaseData?) : State()
+        data class Content(
+            val course: Course,
```

Comment:
Maybe inject course in DI and keep single Content state instead of Idle + Content?